layers/+source-control/version-control/packages.el

```diff
@@ -32,8 +38,188 @@
+      ;; (set-face-foreground 'git-gutter-fr:modified "black")
+      ;; (set-face-foreground 'git-gutter-fr:added    "black")
+      ;; (set-face-foreground 'git-gutter-fr:deleted  "black")
+      (set-face-background 'git-gutter-fr:modified "orange1")
+      (set-face-background 'git-gutter-fr:added    "green4")
+      (set-face-background 'git-gutter-fr:deleted  "red3")
+
+      ;; custom graphics that works nice with half-width fringes
+      (fringe-helper-define 'git-gutter-fr:added nil
+        "..X...."
+        "..X...."
+        "XXXXX.."
+        "..X...."
+        "..X...."
+        )
+      (fringe-helper-define 'git-gutter-fr:deleted nil
+        "......."
+        "......."
+        "XXXXX.."
+        "......."
+        "......."
+        )
+      (fringe-helper-define 'git-gutter-fr:modified nil
+        "..X...."
+        ".XXX..."
+        "XX.XX.."
+        ".XXX..."
+        "..X...."
+        ))))
+
+(defun version-control/init-git-gutter+ ()
+  (use-package git-gutter+
+    :commands global-git-gutter+-mode
+    :if (eq version-control-diff-tool 'git-gutter+)
+    :init
+    ;; If you enable global minor mode
+    (global-git-gutter+-mode t)
+    (setq
+     ;; git-gutter:update-interval 2
+          git-gutter+-modified-sign " "
+          git-gutter+-added-sign "+"
+          git-gutter+-deleted-sign "-"
+          git-gutter+-diff-option "-w"
+          git-gutter+-hide-gutter t
+          ;; git-gutter:ask-p nil
+          ;; git-gutter:verbosity 0
+          ;; git-gutter:handled-backends '(git hg bzr svn)
+          ;; git-gutter:hide-gutter t
+          )
+    (spacemacs|define-micro-state vcs :doc "[n] next [N] previous [r] revert [s] stage [h] show-hunk [D] diff-unstaged [w] stage-file [u] unstage-file [c] commit [C] commit now [f] fetch [F] pull [P] push [l] log [t] toggle margin [q] quit"
+      :disable-evil-leader t
+      :evil-leader "g."
+      :bindings
+      ("C" magit-commit :exit t)
+      ("c" magit-commit-popup :exit t)
+      ("f" magit-fetch-popup :exit t)
+      ("F" magit-pull-popup :exit t)
+      ("P" magit-push-popup :exit t)
+      ("l" magit-log-popup :exit t)
+      ("r" git-gutter+-revert-hunks)
+      ("N" git-gutter+-previous-hunk)
+      ("n" git-gutter+-next-hunk)
+      ("s" git-gutter+-stage-hunks)
+      ("D" magit-diff-unstaged :exit t)
+      ("w" magit-stage-file)
+      ("u" magit-unstage-file)
+      ("h" git-gutter+-show-hunk-inline-at-point)
+      ("t" global-git-gutter+-mode)
+      ("q" nil :exit t))
+    ;; identify magit changes
+    (add-hook 'magit-pre-refresh-hook 'git-gutter+-refresh)
+    (defun version-control/unstage-file ()
+      (interactive)
+      (magit-unstage-file (buffer-file-name))
+      (git-gutter+-refresh)
+      )
+    (spacemacs|hide-lighter git-gutter+-mode)
+    :config
+    (set-face-foreground 'git-gutter+-modified "black")
+    (set-face-foreground 'git-gutter+-added    "black")
+    (set-face-foreground 'git-gutter+-deleted  "black")
```

Comment:
Do you think if it's good idea to set colors explicitly? They might not blend with theme colors. I don't know the right solution but it somehow inheriting from current theme. Anyhow, seems like a personal preference, if it can't be handled, you may ignore.